### PR TITLE
Improve logs collection from resources deloyed during system tests execution

### PR DESCRIPTION
**Pull request by Frawless**
Signed-off-by: Jakub Stejskal <xstejs24@gmail.com>

### Type of change

- Enhancement / new feature
- Refactoring

### Description

THis PR improve log generated by STs execution, fix logs collection from pods used within STs and also archive debug log generated by STs properly on azure.

### Checklist

- [ ] Make sure all tests pass

### CODE REVIEW

**Comment by Frawless** [comment]
Debug log is now properly stored as part of the run - https://dev.azure.com/cncf/strimzi/_build/results?buildId=126079&view=artifacts&pathAsName=false&type=publishedArtifacts . Feel free to check it and check the output of info log from the acceptance build. I would like to also solve log from `CustomResoruces` on debug level but in a separate PR.

File: .azure/templates/steps/system_test_general.yaml [hunk 1]
```diff
@@ -107,7 +107,15 @@ jobs:
...
-        options: '-P${{ parameters.profile }} -DexcludedGroups=flaky,loadbalancer,networkpolicies,${{ parameters.excludedGroups }} -Dmaven.javadoc.skip=true -B -V -Dfailsafe.rerunFailingTestsCount=2 -Djunit.jupiter.execution.parallel.enabled=${{ parameters.run_parallel }} -Djunit.jupiter.execution.parallel.config.fixed.parallelism="${{ parameters.parallel }}"'
+        options: '
+          -P${{ parameters.profile }} 
+          -DexcludedGroups=flaky,loadbalancer,networkpolicies,${{ parameters.excludedGroups }} 
+          -Dmaven.javadoc.skip=true -B -V 
+          -Dfailsafe.rerunFailingTestsCount=2 
+          -Djunit.jupiter.execution.parallel.enabled=${{ parameters.run_parallel }} 
+          -Djunit.jupiter.execution.parallel.config.fixed.parallelism="${{ parameters.parallel }}" 
+          -Dskip.surefire.tests
+          '
```
Comment (scholzj) [inline .azure/templates/steps/system_test_general.yaml:118]: This looks super ugly. Can't you use YAMl block here? (https://yaml-multiline.info/)

File: .azure/templates/steps/system_test_general.yaml [hunk 2]
```diff
@@ -107,7 +107,15 @@ jobs:
         publishJUnitResults: true
         testResultsFiles: '**/failsafe-reports/TEST-*.xml'
         goals: 'verify'
-        options: '-P${{ parameters.profile }} -DexcludedGroups=flaky,loadbalancer,networkpolicies,${{ parameters.excludedGroups }} -Dmaven.javadoc.skip=true -B -V -Dfailsafe.rerunFailingTestsCount=2 -Djunit.jupiter.execution.parallel.enabled=${{ parameters.run_parallel }} -Djunit.jupiter.execution.parallel.config.fixed.parallelism="${{ parameters.parallel }}"'
+        options: '
+          -P${{ parameters.profile }} 
+          -DexcludedGroups=flaky,loadbalancer,networkpolicies,${{ parameters.excludedGroups }} 
+          -Dmaven.javadoc.skip=true -B -V
```
Comment (scholzj) [inline .azure/templates/steps/system_test_general.yaml:113]: Should `-B` and `-V` be on a separate line as well?

**Review by scholzj — APPROVED** [review]
Some nits. But LGTM otherwise I guess if it works.

**Review by im-konge — APPROVED** [review]

File: .azure/templates/steps/system_test_general.yaml (hunk 1 above)
Comment (Frawless) [reply .azure/templates/steps/system_test_general.yaml:118]: Sure, I was thinking about it but I decided to try this at first. Will change it.

File: .azure/templates/steps/system_test_general.yaml (hunk 2 above)
Comment (Frawless) [reply .azure/templates/steps/system_test_general.yaml:113]: Correct

**Pull request merged**